internal/testrunner/runners/system/runner.go

```diff
@@ -443,6 +443,11 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 
 		var err error
 		docs, err = r.getDocs(dataStream)
+
+		if config.Assert.HitCount > 0 {
```

Comment:
@andrewkroh Since I already had context, I added this logic per your guidance. If this case needs more careful handling, I'll have someone else refactor.